full-stack-practice/client/src/Catalog.tsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useState } from 'react';
+import { Product } from '../../server/server.ts';
+import { toDollars } from '../src/lib/index';
+import { Link } from 'react-router-dom';
+import { ClientError } from '../../server/lib/client-error.ts';
+export function Catalog() {
+  const [products, setProducts] = useState<Product[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+
+  useEffect(() => {
+    async function loadItems() {
+      try {
+        const response = await fetch('/api/products');
+        if (!response.ok) {
+          throw new ClientError(400, 'result is invalid');
```

Comment:
`ClientError` is a server-side type only. You should not use it in React code. Just use `Error`.